Source/Fuse.Text/Renderer.uno

```diff
@@ -176,7 +176,8 @@ namespace Fuse.Text
 
 						ClipPosition: Vector.Transform(float4(Position, 0, 1), pixelToClipSpaceMatrix);
 						float4 sampleColor: sample(texture, TexCoord, SamplerState.LinearClamp);
-						PixelColor: grayScale ? color * float4(sampleColor.X) : sampleColor;
+						float4 outColor: grayScale ? color * float4(sampleColor.X) : float4(sampleColor.XYZ, color.W * sampleColor.W);
+						PixelColor: outColor * outColor.W;
```

Comment:
OK, so it looks like there's two "problems" this tries to fix. One is that in the grayScale case `color` is non-premultiplied, which could be better fixed by just pre-multiplying `color` before the shader:
```patch
diff --git a/Source/Fuse.Text/Renderer.uno b/Source/Fuse.Text/Renderer.uno
index 9fd28b53e3..a0bbd21c50 100644
--- a/Source/Fuse.Text/Renderer.uno
+++ b/Source/Fuse.Text/Renderer.uno
@@ -139,6 +139,8 @@ namespace Fuse.Text
 
 		public void Draw(float4 color, float4x4 pixelToClipSpaceMatrix)
 		{
+			color = float4(color.XYZ * color.W, color.W);
+
 			lock (_glyphAtlasMutex)
 			{
 				var batches = GetBatches();
```
The other is that color isn't taken into account in the non-grayScale at all. As I said below, I think this is actually correct, at least in the current definition.

Reply:
If you do this change, then the second change would be reduced to:
```diff
diff --git a/Source/Fuse.Text/Renderer.uno b/Source/Fuse.Text/Renderer.uno
index 9fd28b53e3..2231ffd20a 100644
--- a/Source/Fuse.Text/Renderer.uno
+++ b/Source/Fuse.Text/Renderer.uno
@@ -176,7 +176,7 @@ namespace Fuse.Text
 
 						ClipPosition: Vector.Transform(float4(Position, 0, 1), pixelToClipSpaceMatrix);
 						float4 sampleColor: sample(texture, TexCoord, SamplerState.LinearClamp);
-						PixelColor: grayScale ? color * float4(sampleColor.X) : sampleColor;
+						PixelColor: grayScale ? color * float4(sampleColor.X) : color * sampleColor;
 					};
 				}
 			}
```

Reply:
Also, I don't think the change you propose here is correct either. AFAIK, sampleColor is pre-multiplied, so it's alpha would end up being multiplied in twice in this case. It's probably not easy to see, because sampleColor.W mostly contains anti-aliasing details, but thin details should become even thinner than they should be with this approach.

Reply:
I've pushed the change. I went with a modified form of your first change. Note we only `sampleColor * color.W` since otherwise we'd be affecting the tinting/coloring as well, not just the opacity.